### ISSUE TITLE
Suggest `v.trailing_zeros() == n` over `(v & (1 << n)) != 0`

### ISSUE THREAD

**Issue by mcarton** [rust-lang/rust-clippy#1195]
What it says in the title. There's probably other similar expressions Rust have nice functions for.

See https://stackoverflow.com/questions/39178648/how-to-match-against-a-flags-bitmasks-in-rust.

**Comment by llogiq** [comment]
AFAIR, `trailing_zeroes` does not say anything about the higher bits, though.

**Comment by porglezomp** [comment]
Yeah, these aren't logically the same unless there's the same kind of `if/else if` chain that's in that stack overflow question. `(v & (1 << n)) != 0` is "is the nth bit set?", but `v.trailing_zeros()` is "are all the bits less than n unset?"

**Comment by oli-obk** [comment]
I created a lint for one of the

> other similar expressions Rust has nice functions for

in #1869

**Comment by Jarcho** [comment]
Closing since these aren't equivalent. If someone want to open issues for functions which are missing manual impl lints feel free.
